app/src/main/java/com/example/mpvcalculatorrx/mvp/view/CalculatorView.java

```diff
@@ -0,0 +1,121 @@
+package com.example.mpvcalculatorrx.mvp.view;
+
+import android.app.Activity;
+import android.view.View;
+import android.widget.TextView;
+
+import com.example.mpvcalculatorrx.R;
+import com.example.mpvcalculatorrx.util.bus.RxBus;
+import com.example.mpvcalculatorrx.util.bus.observers.EqualsButtonObserver;
+import com.example.mpvcalculatorrx.util.bus.observers.ResetButtonObserver;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+
+public class CalculatorView extends MvpActivityView {
+
+    @BindView(R.id.text_display) protected TextView textDisplayView;
+
+    public CalculatorView(Activity activity) {
+        super(activity);
+    }
+
+    public void setDisplayValue(final String value) {
+        textDisplayView.setText(value);
+    }
+
+    @OnClick({R.id.button_0, R.id.button_1, R.id.button_2, R.id.button_3, R.id.button_4, R.id.button_5, R.id.button_6, R.id.button_7, R.id.button_8, R.id.button_9})
+    protected void digitButtonPressed(View button) {
+        switch (button.getId()) {
+            case R.id.button_0:
+                digitButtonPressed(0);
```

Comment:
Instead of calling n times your method, save the number to be sent in a variable and then, at the end of the switch call once your method.

Reply:
Done